openmc/mgxs/library.py

```diff
@@ -717,3 +725,307 @@ def load_from_file(filename='mgxs', directory='mgxs'):
+            Defaults to True.
+
+        Returns
+        -------
+        mat_names : Iterable of str
+            Iterable of material names generated during this routine and
+            applies to the cross section library.  Note this is returned if
+            the return_names parameter is provided.
+
+        Raises
+        ------
+        ValueError
+            When the Library object is initialized with insufficient types of
+            cross sections for the Library.
+
+        See also
+        --------
+        Library.dump_to_file(mgxs_lib, filename, directory)
+
+        """
+
+        # Check the provided parameters
+        cv.check_value('xs_type', xs_type, ['macro', 'micro'])
+        if domain_names is not None:
+            cv.check_iterable_type('domain_names', filename, basestring)
+        if xs_ids is not None:
+            if isinstance(xs_ids, basestring):
+                # If we only have a string lets convert it now to a list
+                # of strings.
+                xs_ids = [xs_ids for i in range(len(self.domains))]
+            else:
+                cv.check_iterable_type('xs_ids', xs_ids, basestring)
+        else:
+            xs_ids = ['1g' for i in range(len(self.domains))]
+        cv.check_type('filename', filename, basestring)
+        cv.check_type('directory', directory, basestring)
+
+        # Construct the collection of the nuclides to report
+        if self.by_nuclide:
+            nuclides = self.all_mgxs[1][self.mgxs_types[-1]].get_all_nuclides()
+        else:
+            xs_type = 'macro'
+
+        # Make directory if it does not exist and build our filename
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+        full_filename = os.path.join(directory, filename + '.xml')
+        full_filename = full_filename.replace(' ', '-')
+
+        # Initialize file
+        mgxs_file = openmc.MGXSLibrary(self.energy_groups)
+
+        # Set the scattering order as isotropic until
+        # support for higher orders are included in openmc.mgxs
+        order = 0
+
+        # Build XSdata objects
+        xsdatas = []
+
+        mat_names = {}
+        for i in range(len(self.domains)):
+
+            id = self.domains[i].id
+            if not self.by_nuclide:
+                # Build & add metadata to XSdata object
+                # (Use i here because k in nuclides will add chars to this)
+                if domain_names is None:
+                    name = 'set' + str(i + 1)
+                else:
+                    name = domain_names[i]
+                name += '.' + xs_ids[i]
+                xsdata = openmc.XSdata(name, self.energy_groups)
+                xsdata.order = order
+
+                mat_names[id] = name
+
+                # Now get xs data itself
+                if 'transport' in self.mgxs_types:
+                    if self.correction == 'P0':
+                        xsdata.set_total(self.all_mgxs[id]['transport'],
```

Comment:
I'm not convinced that we need the `subdomains` argument here (and in the `XSdata` class setters). Is this something you've used? At the moment `subdomains` is *only* used to index `'distribcell'` tally filters, though it could might also be used for to index mesh cells if we support mesh domains for `MGXS` mesh domains in the future. It's not clear to me that we have the software infrastructure to pipe MGXS defined on a distribcell mesh back into OpenMC; hence, my understanding is that `subdomains` is not necessary here.